packages/web/server/queries/complex/concentrated-liquidity.ts

```diff
@@ -0,0 +1,526 @@
+        if (current) {
+          sumByDenoms.set(coin.currency.coinMinimalDenom, current.add(coin));
+        } else {
+          sumByDenoms.set(coin.currency.coinMinimalDenom, coin);
+        }
+        return sumByDenoms;
+      }, new Map())
+      .values()
+  );
+}
+
+export async function getTotalEarned({
+  totalSpreadRewards,
+  totalIncentivesRewards,
+  calculatePrice,
+}: {
+  totalSpreadRewards: PositionPerformance["total_spread_rewards"];
+  totalIncentivesRewards: PositionPerformance["total_incentives_rewards"];
+  calculatePrice?: boolean;
+}) {
+  const [spreadRewards, incentivesRewards] = await Promise.all([
+    mapRawAssetsToCoinPretty({
+      rawAssets: totalSpreadRewards,
+      calculatePrice,
+    }),
+    mapRawAssetsToCoinPretty({
+      rawAssets: totalIncentivesRewards,
+      calculatePrice,
+    }),
+  ]);
+
+  const earnedCoinDenomMap = new Map<
+    string,
+    Awaited<ReturnType<typeof mapRawAssetsToCoinPretty>>[number]
+  >();
+  [...spreadRewards, ...incentivesRewards].forEach((coin) => {
+    const existingCoin = earnedCoinDenomMap.get(coin.currency.coinMinimalDenom);
+    if (existingCoin) {
+      earnedCoinDenomMap.set(
+        coin.currency.coinMinimalDenom,
+        existingCoin.add(coin)
+      );
+    } else {
+      earnedCoinDenomMap.set(coin.currency.coinMinimalDenom, coin);
+    }
+  });
+  return Array.from(earnedCoinDenomMap.values());
+}
+
+const clCache = new LRUCache<string, CacheEntry>(DEFAULT_LRU_OPTIONS);
+async function getUserUnbondingPositions({
+  userOsmoAddress,
+}: {
+  userOsmoAddress: string;
+}) {
+  return cachified({
+    cache: clCache,
+    key: `${userOsmoAddress}-cl-unbonding-info`,
+    getFreshValue: async () => {
+      const { positions_with_period_lock } = await queryCLUnbondingPositions({
+        bech32Address: userOsmoAddress,
+      });
+      return positions_with_period_lock;
+    },
+    ttl: 5 * 1000, // 5 seconds
+  });
+}
+
+async function getUserDelegatedClPositions({
+  userOsmoAddress,
+}: {
+  userOsmoAddress: string;
+}) {
+  return cachified({
+    cache: clCache,
+    key: `${userOsmoAddress}-cl-delegated-positions`,
+    getFreshValue: async () => {
+      const { cl_pool_user_position_records } = await queryDelegatedClPositions(
+        {
+          bech32Address: userOsmoAddress,
+        }
+      );
+      return cl_pool_user_position_records;
+    },
+    ttl: 5 * 1000, // 5 seconds
+  });
+}
+
+async function getUserUndelegatingClPositions({
+  userOsmoAddress,
+}: {
+  userOsmoAddress: string;
+}) {
+  return cachified({
+    cache: clCache,
+    key: `${userOsmoAddress}-cl-undelegating-positions`,
+    getFreshValue: async () => {
+      const { cl_pool_user_position_records } =
+        await queryUndelegatingClPositions({
+          bech32Address: userOsmoAddress,
+        });
+      return cl_pool_user_position_records;
+    },
+    ttl: 5 * 1000, // 5 seconds
+  });
+}
+
+export async function normalizePositions({
```

Comment:
I'm not sure what you mean with the function title "normalizePositions". A comment or more descriptive function name would be appreviated